### DISCUSSION TITLE
Python testing using dask and tailscale

### DISCUSSION THREAD

**Discussion by arunoruto** [dask/dask#10911] [Q&A]
I built a setup using Dask to run my Python on multiple servers. This wasn't that easy since my code needs a GPU, so I resorted to the rapids.ai container!

Now, I was thinking about how to test my code. I would like to do this using a GitHub workflow, but the computations are expensive, and the GitHub runners have no GPUs. On the other hand, I already have a working setup at my lab.

So my question is: Is there a setup that enables me to run tests using Dask? It would be awesome if multiple servers could execute multiple test files in parallel! I was searching online, but haven't found much information on it (or I was googling it wrong).
One can use the tailscale GitHub action to connect to a virtual network where the scheduler is and order it to run the tests, bypassing the need for a self-hosted runner, which can be a bit cumbersome.

**Comment by mrocklin** [comment]
I think that @mariusvniekerk had something at one point

**Reply by mariusvniekerk** [reply]
Yeah I wrote pytest-dask years ago. 

Given the amount of rather scary stuff altering the pytest serialization behavior it does I highly doubt it.  

It might inspire a more modern take on the idea.

Would be happy to transfer over the pypi name if you feel like writing a modern equivalent and get something working.

**Reply by arunoruto** [reply]
Thanks for the information! I guess [this](https://github.com/mariusvniekerk/pytest-dask) is the repository for pytest-dask so I will look at it! I am not that knowledgeable with pytest, so I will need to read into it a bit, but the idea sounds promising and would solve the issues I am having with the self-hosted runners!
